src/platform/networking/common/networking.ts

```diff
@@ -226,13 +230,48 @@ export interface IChatEndpoint extends IEndpoint {
 }
 
 /** Function to create a standard request body for CAPI completions */
-export function createCapiRequestBody(model: string, options: ICreateEndpointBodyOptions) {
+export function createCapiRequestBody(options: ICreateEndpointBodyOptions, model: string) {
 	// FIXME@ulugbekna: need to investigate why language configs have such stop words, eg
 	// python has `\ndef` and `\nclass` which must be stop words for ghost text
 	// const stops = getLanguageConfig<string[]>(accessor, ConfigKey.Stops);
 
+	const messages = [];
+	for (const message of options.messages) {
+		if (message.role === Raw.ChatRole.Assistant) {
+			let newMessage = rawMessageToCAPI(message);
+			for (const content of message.content) {
+				// opaque type
+				if (content.type === 2) {
+					const data = content.value as {
+						type: string;
+						thinking: ThinkingData;
+					};
+					if (data.type === 'thinking' && data.thinking.id) {
+						if (options.reasoningPropertyType === 'AzureOpenAI') {
```

Comment:
If the goal is to say which endpoint it is, then this could be done in `createRequestBody` for the endpoint without passing a new property around